### PR TITLE
Hds 1814 refactor navmenu

### PR DESCRIPTION
**Pull request by harriplappalainen**
## Description

Refactor so there isn't duplicate code in Header.NavigationMenu and Header.ActionBarNavigationMenu.

Also fix UI spacing in nested mobile navigation menu: there doesn't need to be space between dropdown button and right hand corner.

## Related Issue

https://helsinkisolutionoffice.atlassian.net/browse/HDS-1814

## How Has This Been Tested?
Locally, tests
[Demo](https://city-of-helsinki.github.io/hds-demo/refactor-navmenu?path=/story/components-header--with-full-features)

### CODE REVIEW

File: packages/react/src/components/header/components/navigationLink/NavigationLink.tsx [hunk 1]
```diff
@@ -191,6 +196,7 @@ export const NavigationLink = <T extends React.ElementType = 'a'>({
       }
     }
   };
+  console.log(dropdownButtonClassName);
```
Comment (mikkojamG) [inline packages/react/src/components/header/components/navigationLink/NavigationLink.tsx:199]: I'm guessing this was left here unintentionally? 🤔

File: packages/react/src/components/header/components/navigationLink/navigationLinkDropdown/NavigationLinkDropdown.tsx [hunk 2]
```diff
@@ -84,12 +89,12 @@ export const NavigationLinkDropdown = ({
   };
 
   const childElements = getChildElementsEvenIfContainersInbetween(children);
-
+  console.log(dropdownButtonClassName);
```
Comment (mikkojamG) [inline packages/react/src/components/header/components/navigationLink/navigationLinkDropdown/NavigationLinkDropdown.tsx:92]: I'm guessing this was left here unintentionally? 🤔

File: packages/react/src/components/header/components/navigationLink/NavigationLink.tsx (hunk 1 above)
Comment (harriplappalainen) [reply packages/react/src/components/header/components/navigationLink/NavigationLink.tsx:199]: Yes 🤦 Thanks for finding this!

File: packages/react/src/components/header/components/navigationLink/navigationLinkDropdown/NavigationLinkDropdown.tsx (hunk 2 above)
Comment (harriplappalainen) [reply packages/react/src/components/header/components/navigationLink/navigationLinkDropdown/NavigationLinkDropdown.tsx:92]: Yes.. 🤦

**Review by mikkojamG — APPROVED** [review]
LGTM 👍

**Pull request merged**